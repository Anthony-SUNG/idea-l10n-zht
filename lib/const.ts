import { join } from 'upath2';
import { __root } from '../test/__root';

export const __plugin_downloaded_dir = join(__root, 'original-plugin');
export const __plugin_downloaded_dir_unzip = join(__root, 'original-plugin-raw');

export const __plugin_dev_dir = join(__root, 'plugin-dev-raw');
