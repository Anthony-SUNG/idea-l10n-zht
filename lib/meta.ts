export function updateMeta(content_new: string)
{
	return content_new
		.replace(/<name>.+<\/name>/, `<name>Chinese (Traditional) Language Pack / 繁體中文語言包</name>`)
		.replace(/<vendor>.+<\/vendor>/, `<vendor>bluelovers</vendor>`)
		.replace(/<description>.+<\/description>/, `<description><![CDATA[Jetbrains IntelliJ Chinese (Traditional) Language Pack / 繁體中文語言包<p>The Chinese Language Pack localizes the UI of IntelliJ IDEA, AppCode, CLion, DataGrip, GoLand, PyCharm, PhpStorm, RubyMine, and WebStorm into Chinese.</p><p>中文語言包將為您的 IntelliJ IDEA, AppCode, CLion, DataGrip, GoLand, PyCharm, PhpStorm, RubyMine, 和WebStorm 帶來中文化的界面。</p><h2>install</h2><ul><li>install from <a href="https://github.com/bluelovers/idea-l10n-zht/raw/master/plugin-dev-out/zh.jar" target="_blank">github</a></li><li>install from <a href="https://plugins.jetbrains.com/plugin/18365-chinese-traditional-language-pack-----">jetbrains</a></li></ul><blockquote><p>注意：請先移除或禁用<a href="https://plugins.jetbrains.com/plugin/13710-chinese-simplified-language-pack----" target="_blank">官方簡體中文語言包</a></p></blockquote>]]></description>`)
		.replace(/<id>.+<\/id>/, `<id>com.intellij.zh-tw</id>`)
		.replace(/\n+$/, '')
		;
}
