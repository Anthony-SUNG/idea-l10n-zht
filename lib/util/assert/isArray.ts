import { ITSArrayListMaybeReadonly } from 'ts-type/lib/type/base';

export function isArray<T extends ITSArrayListMaybeReadonly<any>>(value: T | unknown): value is T
{
	return Array.isArray(value)
}
