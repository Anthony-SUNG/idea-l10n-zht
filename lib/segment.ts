import { getSegment, processText } from "novel-segment-cli";
import { chkcrlf, CR, CRLF, LF } from 'crlf-normalize';
import { EnumDictDatabase } from "novel-segment/lib/const";
import Bluebird from 'bluebird';

let inited: boolean;

export function initIdeaSegmentText()
{
	return Bluebird.resolve()
		.then(() =>
		{
			return !inited && getSegment().then(segment =>
			{

				let db_dict = segment.getDictDatabase(EnumDictDatabase.TABLE);
				let db_synonym = segment.getDictDatabase(EnumDictDatabase.SYNONYM);

				db_dict
					.add(['選項卡', 0x100000, 0])
				;

				db_synonym
					.add(['頁籤', '選項卡'])
					.add(['視窗', '窗口'])
					.add(['默認', '預設'])
					.add(['列印', '打印'])
				;

				inited = true;
			})
		})
}

export function processIdeaSegmentText(text: string)
{
	let _lb = chkcrlf(text);

	return initIdeaSegmentText().then(() => processText(text, {
		convertToZhTw: true,
		crlf: _lb.crlf ? CRLF : (_lb.lf || !_lb.cr) ? LF : CR,
	})).then(text => {
		return text
			//.replace(/打印/, '列印')
	})
}
