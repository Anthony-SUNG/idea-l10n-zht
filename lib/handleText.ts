import { processIdeaSegmentText } from './segment';
import { updateMeta } from './meta';
import { textToRegexp } from './util/text-to-regexp';

export function handleText(content_old: string, info?: {
	file?: string,
})
{
	const file = info?.file ?? '';

	return processIdeaSegmentText(content_old)
		.then(content_new =>
		{
			if (/META-INF\/plugin\.xml$/i.test(file))
			{
				content_new = updateMeta(content_new);
			}
			else if (/(?:Git(?:hub)?|Vcs(?:Log)?|Svn|Diff)Bundle\.properties$/i.test(file))
			{
				content_new = content_new
					.replace(/提取/g, '獲取')
					.replace(/儲存庫/g, '版本庫')
					.replace(/統一/g, '統合')
				;
			}
			else if (/LeakableMapKey\.html$/i.test(file))
			{
				content_new = content_new
					.replace(textToRegexp(/卸載|移除/g), '解除安裝')
				;
			}

			return content_new
		})
}
