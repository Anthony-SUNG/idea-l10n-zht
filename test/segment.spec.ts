import { lazyMatchSynonym001, lazyMatchSynonym001Not } from '@novel-segment/assert';
import { processIdeaSegmentText } from '../lib/segment';
import { ITSValueOrArrayMaybeReadonly } from 'ts-type/lib/type/base';

jest.setTimeout(60 * 1000);

type ITestList = [ITSValueOrArrayMaybeReadonly<string>, string][];

/**
 * 檢查是否確實轉換
 * 通過測試不代表轉換符合預期，因為檢查詞必須要正確設定才可以
 */
describe(`segment`, () =>
{

	/**
	 * 繁體/簡體皆可 但簡體比較能反映真實運作
	 */
	(<ITestList>[
		//[`繁體`, `繁體/簡體皆可 但簡體比較能反映真實運作`],

		[`打印`, `action.Print.text=打印(_P)…\naction.Print.description=打印文件`],
		[`粘貼`, `從剪貼板粘貼`],
		[`剪貼板`, `從剪貼板粘貼`],
		[`選項卡`, `編輯器選項卡`],
		[`選項卡`, `action.close.all.unmodified.editors=關閉未修改選項卡(_U)`],
		[`只讀`, `action.ToggleReadOnlyAttribute.text=切換只讀特性`],
		[`標籤頁`, `關閉組內未固定標籤頁`],
		[`標簽頁`, `關閉組內未固定標簽頁`],
		[`文件夾`, `，右击任何文件夹，`],
		[`文件夾`, `仅报告位於源文件夹下的空目录`],
		[`全局`, `使工具全局可见(&B)`],
		[`全局库`, `导入的项目引用了未知的全局库`],
		[`服務器`, `以下文件被禁止，因为其中一个文件很可能导致服务器崩溃。\n{0}`],
		[`屏幕`, `action.android.emulator.home.button.text=主屏幕`],
		[`搜索`, `正在搜索受影響的檔案路徑…`],
		[`查找`, `查找下一个/移至下一个匹配项(_N)…`],
		[`宏`, `无法解析宏 ''{1}'' 的形参 ''{0}''`],
		[`宏`, `查看、更改、录制、播放宏`],
		[`形參`, `類型形參 ''{0}'' 不能直接實例化`],
		[`應用程序`, `重新加載應用程序設定`],
		[`字符串`, `在活動編輯器中尋找字符串`],
		[`反方向`, `反方向重複上一次尋找`],
		[`圖像`, `圖像已複製到剪貼簿`],
		[`信息`, `顯示詳細信息`],
		[`數據源`, `外部數據源`],
		[`數據`, `刷新測試管理數據`],
		[`代碼`, `檔案和代碼模板`],
		[`變量`, `在當前範圍內未初始化或不可用的變量將不會顯示`],
		[`類型作為`, `預期布爾類型作為方法 {0} 返回類型`],
		[`布爾`, `預期布爾類型作為方法 {0} 返回類型`],
		[`布爾值`, `首選顯式 '-n' 檢查非空字串。使用 '=' 或 '-ne' 檢查布爾值和整數。`],
		[`運行`, `關閉具有運行進程的工具視窗時:`],
		[`激活`, `激活工具窗口: {0}`],
		[`窗口`, `激活工具窗口: {0}`],
		[`用戶`, `與以下用戶完全同步:`],
		[`項目`, `<html><body>您想在哪裡打開項目 ''{0}''?</body></html>`],
		[`加載`, `導入最適合乾淨項目。導入未被清理的項目時，重新加載過程會忽略包含最新目標的源文件。`],
		[`插件`, `Kotlin 插件有可用的新版本 {0}。<b><a href="#">安裝</a></b>`],
		[`兼容`, `從右到左文本兼容性問題`],
		[`光標`, `從文本光標開始的所有內容`],
		[`標識符`, `不允許在一個模板中混用駝峰拼寫法和小寫標識符`],
		[`骆驼拼写法`, `使用骆驼拼写法样式的方法名，没有 "get" 前缀，第一个字符大写。 例如，属性名称 '_my_property' 被转换为 'MyProperty'。`],
		[
			`駝峰拼寫法`,
			`<p><a href="https://kotlinlang.org/docs/coding-conventions.html#function-names">建議的命名慣例</a>：必須以小寫字母開頭，使用駝峰拼寫法並且沒有下划線。</p>`,
		],
		[`導出`, `選擇包含導出的檢查結果的目錄`],
		[`導入`, `選擇靜態可導入類`],
		[`緩存`, `沒有緩存，初始編制索引時間會顯著增加`],
		[`倉庫`, `選擇要合併的倉庫`],
		[`常量`, `接口中的常量必須具有初始值設定項`],
		[`界面`, `新的非模式提交界面可用。从工具窗口提交并在编辑器中查看差异。`],
		[`隊列`, `消息隊列連接(彈出視窗)`],
		[`掩碼`, `文件掩码(&F)`],

	]).forEach(text =>
	{

		test(_handleTitles(text), async () =>
		{

			let actual = await processIdeaSegmentText(text[1]);

			lazyMatchSynonym001Not(actual, [text[0]].flat());
			//expect(actual).not.toContain(text[0])
			expect(actual).toMatchSnapshot();

		});

	});

})

/**
 * 檢查是否發生誤轉換
 */
describe(`should include`, () =>
{

	(<ITestList>[

		[`顯示`, `顯示屏幕外圖像`],
		[`頁面`, `打开文件或项目。您也可以通过拖放到欢迎屏幕来打开项目或编辑文件。`],
		[`頁面`, `显示初始屏幕`],
		[`原始碼`, `选择根源代码`],
		[`全域類別庫`, `导入的项目引用了未知的全局库`],
		[`執行巨集`, `查看、更改、录制、播放宏`],
		[`駝峰式命名`, `<option name="切换骆驼拼写法" path="ActionManager" hit="切换骆驼拼写法/蛇形拼写法" />`],

		[`位元`, `不兼容的按位掩码运算`],

		[`導向`, `用于将面向对象的域模型映射到关系资料库的框架。`],

		[`常數`, `闭包使用外部非常量变数`],

		[`宣告`, `导航到上一个声明`],
		[`副檔名`, `找不到档案扩展名 ''{0}'' 的脚本引擎`],
		[`執行緒`, `框架不可用于未挂起的缐程`],
		[`堆疊`, `合并相同的堆栈跟踪`],
		[`運算子`, `操作符置于下一行`],
		[`右鍵選單`, `，然后从上下文菜单中选择“调试器”。`],
		[`物件`, `，对象`],
		[`標頭檔`, `，创建新的 Objective-C 类和头文件`],
		[`標頭檔`, `，创建新的 Objective-C 类和头档案`],
		[`整數型`, `使用非整型表达式 {1}{2} 指定了{0, choice, 0#宽度|1#精度|2#谓词}`],

		[`元素`, `單元素數組`],
		[`元素`, `最大元素數`],

		[`排程器`, `找不到进程调度程序的可用套接字连接埠`],

		[`萬用字元`, `可以使用有界通配符`],
		[`蛇形命名`, `常量和函数时报告使用蛇形拼写法而不是骆驼拼写法的情况`],

		[`執行緒`, `获取线程转储`],

		[`運行時版本`, `about.box.jre=运行时版本: {0} {1}`],
		[`構建`, `about.box.build.number=内部版本号 #{0}`],

	]).forEach(text =>
	{

		test(_handleTitles(text), async () =>
		{

			let actual = await processIdeaSegmentText(text[1]);

			lazyMatchSynonym001(actual, [text[0]].flat());
			//expect(actual).toContain(text[0])
			expect(actual).toMatchSnapshot();

		});

	});

})

function _handleTitles(actual: [ITSValueOrArrayMaybeReadonly<string>, string])
{
	let arr = actual.slice();
	arr[0] = [arr[0]].flat().join('／');
	return arr.join(' - ')
}
