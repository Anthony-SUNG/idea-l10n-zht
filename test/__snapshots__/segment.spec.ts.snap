// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`segment 代碼 - 檔案和代碼模板 1`] = `"檔案和程式碼模板"`;

exports[`segment 信息 - 顯示詳細信息 1`] = `"顯示詳細資訊"`;

exports[`segment 倉庫 - 選擇要合併的倉庫 1`] = `"選擇要合併的儲存庫"`;

exports[`segment 光標 - 從文本光標開始的所有內容 1`] = `"從文本游標開始的所有內容"`;

exports[`segment 全局 - 使工具全局可见(&B) 1`] = `"使工具全域可見(&B)"`;

exports[`segment 全局库 - 导入的项目引用了未知的全局库 1`] = `"匯入的專案引用了未知的全域類別庫"`;

exports[`segment 兼容 - 從右到左文本兼容性問題 1`] = `"從右到左文本相容性問題"`;

exports[`segment 剪貼板 - 從剪貼板粘貼 1`] = `"從剪貼簿貼上"`;

exports[`segment 加載 - 導入最適合乾淨項目。導入未被清理的項目時，重新加載過程會忽略包含最新目標的源文件。 1`] = `"匯入最適合乾淨專案。匯入未被清理的專案時，重新載入過程會忽略包含最新目標的源文件。"`;

exports[`segment 反方向 - 反方向重複上一次尋找 1`] = `"反向重複上一次尋找"`;

exports[`segment 只讀 - action.ToggleReadOnlyAttribute.text=切換只讀特性 1`] = `"action.ToggleReadOnlyAttribute.text=切換唯讀特性"`;

exports[`segment 圖像 - 圖像已複製到剪貼簿 1`] = `"圖片已複製到剪貼簿"`;

exports[`segment 字符串 - 在活動編輯器中尋找字符串 1`] = `"在活動編輯器中尋找字串"`;

exports[`segment 宏 - 无法解析宏 ''{1}'' 的形参 ''{0}'' 1`] = `"無法解析巨集 ''{1}'' 的參數 ''{0}''"`;

exports[`segment 宏 - 查看、更改、录制、播放宏 1`] = `"查看、更改、錄製、執行巨集"`;

exports[`segment 導入 - 選擇靜態可導入類 1`] = `"選擇靜態可匯入類"`;

exports[`segment 導出 - 選擇包含導出的檢查結果的目錄 1`] = `"選擇包含匯出的檢查結果的目錄"`;

exports[`segment 屏幕 - action.android.emulator.home.button.text=主屏幕 1`] = `"action.android.emulator.home.button.text=主螢幕"`;

exports[`segment 布爾 - 預期布爾類型作為方法 {0} 返回類型 1`] = `"預期布林型別作為方法 {0} 返回類型"`;

exports[`segment 布爾值 - 首選顯式 '-n' 檢查非空字串。使用 '=' 或 '-ne' 檢查布爾值和整數。 1`] = `"首選顯式 '-n' 檢查非空字串。使用 '=' 或 '-ne' 檢查布林值和整數。"`;

exports[`segment 常量 - 接口中的常量必須具有初始值設定項 1`] = `"接口中的常數必須具有初始值設定項"`;

exports[`segment 形參 - 類型形參 ''{0}'' 不能直接實例化 1`] = `"類型參數 ''{0}'' 不能直接實例化"`;

exports[`segment 應用程序 - 重新加載應用程序設定 1`] = `"重新載入應用程式設定"`;

exports[`segment 打印 - action.Print.text=打印(_P)…
action.Print.description=打印文件 1`] = `
"action.Print.text=列印(_P)…
action.Print.description=列印檔案"
`;

exports[`segment 掩碼 - 文件掩码(&F) 1`] = `"檔案遮罩(&F)"`;

exports[`segment 插件 - Kotlin 插件有可用的新版本 {0}。<b><a href="#">安裝</a></b> 1`] = `"Kotlin 套件有可用的新版本 {0}。<b><a href=\\"#\\">安裝</a></b>"`;

exports[`segment 搜索 - 正在搜索受影響的檔案路徑… 1`] = `"正在搜尋受影響的檔案路徑…"`;

exports[`segment 數據 - 刷新測試管理數據 1`] = `"刷新測試管理資料"`;

exports[`segment 數據源 - 外部數據源 1`] = `"外部資料來源"`;

exports[`segment 文件夾 - 仅报告位於源文件夹下的空目录 1`] = `"僅報告位於源資料夾下的空目錄"`;

exports[`segment 文件夾 - ，右击任何文件夹， 1`] = `"，右擊任何資料夾，"`;

exports[`segment 服務器 - 以下文件被禁止，因为其中一个文件很可能导致服务器崩溃。
{0} 1`] = `
"以下檔案被禁止，因為其中一個檔案很可能導致伺服器崩潰。
{0}"
`;

exports[`segment 查找 - 查找下一个/移至下一个匹配项(_N)… 1`] = `"尋找下一個/移至下一個符合項(_N)…"`;

exports[`segment 標簽頁 - 關閉組內未固定標簽頁 1`] = `"關閉組內未固定頁籤"`;

exports[`segment 標籤頁 - 關閉組內未固定標籤頁 1`] = `"關閉組內未固定頁籤"`;

exports[`segment 標識符 - 不允許在一個模板中混用駝峰拼寫法和小寫標識符 1`] = `"不允許在一個模板中混用駝峰式命名和小寫關鍵字"`;

exports[`segment 激活 - 激活工具窗口: {0} 1`] = `"啟用工具視窗: {0}"`;

exports[`segment 用戶 - 與以下用戶完全同步: 1`] = `"與以下使用者完全同步:"`;

exports[`segment 界面 - 新的非模式提交界面可用。从工具窗口提交并在编辑器中查看差异。 1`] = `"新的非強制回應提交介面可用。從工具視窗提交並在編輯器中查看差異。"`;

exports[`segment 窗口 - 激活工具窗口: {0} 1`] = `"啟用工具視窗: {0}"`;

exports[`segment 粘貼 - 從剪貼板粘貼 1`] = `"從剪貼簿貼上"`;

exports[`segment 緩存 - 沒有緩存，初始編制索引時間會顯著增加 1`] = `"沒有快取，初始編制索引時間會顯著增加"`;

exports[`segment 變量 - 在當前範圍內未初始化或不可用的變量將不會顯示 1`] = `"在當前範圍內未初始化或不可用的變數將不會顯示"`;

exports[`segment 運行 - 關閉具有運行進程的工具視窗時: 1`] = `"關閉具有執行進程的工具視窗時:"`;

exports[`segment 選項卡 - action.close.all.unmodified.editors=關閉未修改選項卡(_U) 1`] = `"action.close.all.unmodified.editors=關閉未修改頁籤(_U)"`;

exports[`segment 選項卡 - 編輯器選項卡 1`] = `"編輯器頁籤"`;

exports[`segment 隊列 - 消息隊列連接(彈出視窗) 1`] = `"訊息佇列連接(彈出視窗)"`;

exports[`segment 項目 - <html><body>您想在哪裡打開項目 ''{0}''?</body></html> 1`] = `"<html><body>您想在哪裡打開專案 ''{0}''?</body></html>"`;

exports[`segment 類型作為 - 預期布爾類型作為方法 {0} 返回類型 1`] = `"預期布林型別作為方法 {0} 返回類型"`;

exports[`segment 駝峰拼寫法 - <p><a href="https://kotlinlang.org/docs/coding-conventions.html#function-names">建議的命名慣例</a>：必須以小寫字母開頭，使用駝峰拼寫法並且沒有下划線。</p> 1`] = `"<p><a href=\\"https://kotlinlang.org/docs/coding-conventions.html#function-names\\">建議的命名慣例</a>：必須以小寫字母開頭，使用駝峰式命名並且沒有下划線。</p>"`;

exports[`segment 骆驼拼写法 - 使用骆驼拼写法样式的方法名，没有 "get" 前缀，第一个字符大写。 例如，属性名称 '_my_property' 被转换为 'MyProperty'。 1`] = `"使用駝峰式命名樣式的方法名，沒有 \\"get\\" 前綴，第一個字符大寫。 例如，屬性名稱 '_my_property' 被轉換為 'MyProperty'。"`;

exports[`should include 位元 - 不兼容的按位掩码运算 1`] = `"不相容的位元遮罩運算"`;

exports[`should include 元素 - 單元素數組 1`] = `"單元素數組"`;

exports[`should include 元素 - 最大元素數 1`] = `"最大元素數"`;

exports[`should include 全域類別庫 - 导入的项目引用了未知的全局库 1`] = `"匯入的專案引用了未知的全域類別庫"`;

exports[`should include 副檔名 - 找不到档案扩展名 ''{0}'' 的脚本引擎 1`] = `"找不到檔案副檔名 ''{0}'' 的腳本引擎"`;

exports[`should include 原始碼 - 选择根源代码 1`] = `"選擇根原始碼"`;

exports[`should include 右鍵選單 - ，然后从上下文菜单中选择“调试器”。 1`] = `"，然後從右鍵選單中選擇“調試器”。"`;

exports[`should include 執行巨集 - 查看、更改、录制、播放宏 1`] = `"查看、更改、錄製、執行巨集"`;

exports[`should include 執行緒 - 框架不可用于未挂起的缐程 1`] = `"框架不可用於未掛起的執行緒"`;

exports[`should include 堆疊 - 合并相同的堆栈跟踪 1`] = `"合併相同的堆疊跟踪"`;

exports[`should include 宣告 - 导航到上一个声明 1`] = `"導航到上一個宣告"`;

exports[`should include 導向 - 用于将面向对象的域模型映射到关系资料库的框架。 1`] = `"用於將物件導向的域模型映射到關係資料庫的框架。"`;

exports[`should include 常數 - 闭包使用外部非常量变数 1`] = `"閉包使用外部非常數變數"`;

exports[`should include 排程器 - 找不到进程调度程序的可用套接字连接埠 1`] = `"找不到進程排程器的可用套接字連接埠"`;

exports[`should include 整數型 - 使用非整型表达式 {1}{2} 指定了{0, choice, 0#宽度|1#精度|2#谓词} 1`] = `"使用非整數型表達式 {1}{2} 指定了{0, choice, 0#寬度|1#精度|2#謂詞}"`;

exports[`should include 標頭檔 - ，创建新的 Objective-C 类和头文件 1`] = `"，創建新的 Objective-C 類和標頭檔"`;

exports[`should include 標頭檔 - ，创建新的 Objective-C 类和头档案 1`] = `"，創建新的 Objective-C 類和標頭檔"`;

exports[`should include 物件 - ，对象 1`] = `"，物件"`;

exports[`should include 萬用字元 - 可以使用有界通配符 1`] = `"可以使用有界萬用字元"`;

exports[`should include 運算子 - 操作符置于下一行 1`] = `"運算子置於下一行"`;

exports[`should include 頁面 - 打开文件或项目。您也可以通过拖放到欢迎屏幕来打开项目或编辑文件。 1`] = `"打開檔案或專案。您也可以通過拖放到歡迎頁面來打開專案或編輯檔案。"`;

exports[`should include 頁面 - 显示初始屏幕 1`] = `"顯示初始頁面"`;

exports[`should include 顯示 - 顯示屏幕外圖像 1`] = `"顯示螢幕外圖片"`;

exports[`should include 駝峰式命名 - <option name="切换骆驼拼写法" path="ActionManager" hit="切换骆驼拼写法/蛇形拼写法" /> 1`] = `"<option name=\\"切換駝峰式命名\\" path=\\"ActionManager\\" hit=\\"切換駝峰式命名/蛇形拼寫法\\" />"`;
